### ISSUE TITLE
How do you do many to many with more than 2 tables ?

### ISSUE THREAD

**Issue by cavemancoder999** [dpgaspar/Flask-AppBuilder#622]
I have a table for Logs that contains various information about employees
ex:

```
class Log(Model):
        division_id = Column(Integer, ForeignKey('division.id'), nullable=False)
        division = relationship("Division")

        employee_id = Column(Integer, ForeignKey("employee.id"), nullable=False)
        employee = relationship("Employee") 
    
        skill_id = Column(Integer, ForeignKey("skill.id"), nullable=False)
        skill = relationship("Skill")
    
        message = Column(String, default='OK', nullable=False)
        date = Column(DateTime, default=NowTime(), nullable=True)
```

Employee and Skill Tables look like this:

```
 class Employee(Model):
        id = Column(Integer, primary_key=True)
        name = Column(String, unique=True, nullable=False)
        division_id = Column(Integer, ForeignKey('division.id'), nullable=False)
        division = relationship("Division")

 class Skill(Model):
        id = Column(Integer, primary_key=True)
        name = Column(String, unique=True, nullable=False)
```

I am currently using Flask-Appbuilder and I have a Skill View that displays all the logs for the current selected skill. 

 ```
 class LogView(ModelView):
        datamodel = SQLAInterface(Log)
        list_columns = ['division', 'employee', 'skill', 'message', 'date']
        show_template = 'appbuilder/general/model/show_cascade.html'

class SkillLogView(ModelView):
        datamodel = SQLAInterface(Skill)
        list_columns = ['name']
        related_views = [LogView]
        show_template = 'appbuilder/general/model/show_cascade.html'
```
    

In the SkillLogView, I also want to display a list of employee names that have this skill.

How do I also get the employees from the logs that pertain to the current skill ?

I am not sure how to do it, but I thought it might be a case for many to many. The problem is that there are 3 tables, not 2.

Is there a way to do many to many with more than 2 tables ?

Or is there another way to accomplish what I want to do ?

Any help appreciated.

**Comment by cavemancoder999** [comment]
There is an example online that does this very thing!

It has a person table, car table, car ownership table.

All I had to do was substitute employee for person, skill for car, and log for car ownership.

Reference URL: 
https://stackoverflow.com/questions/38248415/many-to-many-with-association-object-and-all-relationships-defined-crashes-on-de

Updated Code:

    class Log(Model):
        division_id = Column(Integer, ForeignKey('division.id'), nullable=False)
        division = relationship("Division")
    
        employee_id = Column(Integer, ForeignKey("employee.id"), nullable=False)
        employee = relationship("Employee", backref=backref('log', passive_deletes='all'))
    
        skill_id = Column(Integer, ForeignKey("skill.id"), nullable=False)
        skill = relationship("Skill", backref=backref('log', passive_deletes='all'))
    
        message = Column(String, default='OK', nullable=False)
        date = Column(DateTime, default=NowTime(), nullable=True)


    class Skill(Model):
        id = Column(Integer, primary_key=True)
        name = Column(String, unique=True, nullable=False)

        employees = relationship('Employee', secondary='log', backref='skill')


    class Employee(Model):
        id = Column(Integer, primary_key=True)
        name = Column(String, unique=True, nullable=False)
    
        division_id = Column(Integer, ForeignKey('division.id'), nullable=False)
        division = relationship("Division")
    
        skills = relationship('Skill', secondary='log', backref='employee')

**Comment by cavemancoder999** [comment]
I have noticed in my SkillLogView that the log count number is correctly displayed, but the employee count number is not. 

The employee count number is currently the same as the log count number.
This is either a bug, or has to do with the way I implemented the table model relationships.

Here is my View
```
class SkillLogView(ModelView):
    datamodel = SQLAInterface(Skill)
    list_columns = ['name']
    related_views = [LogView, EmployeeView]
    show_template = 'appbuilder/general/model/show_cascade.html'
    show_fieldsets = [
        ('Summary', {'fields': ['name']}),
    ]
```
There are 3 logs from 2 users. The log and user entries are correctly listed, but the count on employees is wrong.
![error_record_count](https://user-images.githubusercontent.com/32503910/31263326-7aa23b4a-aa9c-11e7-995d-7bf01b630f14.png)
